### PR TITLE
PM-11123: Notification Status Details view

### PR DESCRIPTION
**Pull request by mzieniukbw**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11123

## 📔 Objective

Get Notifications with Statuses as one database query

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Sql/NotificationCenter/dbo/Views/NotificationStatusDetailsView.sql
Comment (withinfocus) [inline src/Sql/NotificationCenter/dbo/Views/NotificationStatusDetailsView.sql:13]: ⛏️ Newlines at the end.

File: src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs
Comment (withinfocus) [inline src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: ❓ This is here just to aid in querying right? Can we restructure the join usage to avoid this being defined in the view and response objects? I find it confusing to be there when it would be set as the above `UserId`.

Comment (Thomas-Avery) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: What about the case of a notification that is targeted at an organization? 

```
Id : 61c208d3-74ae-4d5f-9c92-150ab941d262
ClientType: All
UserId: NULL
OrganizationId: 44674c0f-a6af-4c09-9c65-03ab89094764
NotificationStatusUserId: <target-userId>
ReadDate: 2025-09-18 19:00:52.0000000
DeletedDate: 2025-09-18 19:00:52.0000000
```

Comment (withinfocus) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: I thought about that, but ... does the caller care? A user is calling this. We could arguably drop both user ID properties.

File: src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs
Comment (Thomas-Avery) [inline src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:1]: The pattern is to put entities that match DB tables in `/Core/<team>/Entities`. 

This is a data model for a view which typically would go in `/Core/NotificationCenter/Models/Data`

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]

File: src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs
Comment (Thomas-Avery) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: True, in these cases it's implied that the notification is for the given user.

Comment (mzieniukbw) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: I can drop it from the `NotificationStatusDetails.cs`. Like you Both said, it does not matter.
But it does from `NotificationStatusDetailsView` and `Notification_ReadByUserIdAndStatus`.
We need to filter on `NotificationStatus.UserId`, so we only return current user statuses. This needs to be done from stored procedure, so the view have to expose it.
We also need `Notification.UserId`, so we only return notifications that match current user (assuming it's not null).
Both are nullable. I just can't see how we can do this if we only have one or another (or combined with ISNULL, also does not work).
Alternatively I can get rid of the `NotificationStatusDetailsView`, that solves this problem.

Comment (withinfocus) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: The easiest path forward might be to keep these in the model and view then, but I don't want them to be seen in the API response model at least.

Comment (mzieniukbw) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:23]: I think it's fine to delete the `NotificationStatusUserId` field from `NotificationStatusDetails.cs` - so the only occurrence is in SQL files.

File: src/Sql/NotificationCenter/dbo/Views/NotificationStatusDetailsView.sql
Comment (mzieniukbw) [reply src/Sql/NotificationCenter/dbo/Views/NotificationStatusDetailsView.sql]: Fixed in 74b1fcdd

File: src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs
Comment (mzieniukbw) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs]: Fixed in 5c936d09

File: src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs
Comment (mzieniukbw) [reply src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs:1]: Fixed in 88e1edd5

**Review by withinfocus — COMMENTED** [review]
I am jumping on a plane so I'll let @Thomas-Avery get final review in.

**Review by withinfocus — APPROVED** [review]
(but my part seems good, and we'll see the real changes next)

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Pull request merged**
